src/main/java/seedu/duke/commands/FindCommand.java

```diff
@@ -0,0 +1,35 @@
+package seedu.duke.commands;
+
+import seedu.duke.storage.Storage;
+import seedu.duke.task.Task;
+import seedu.duke.task.TaskList;
+import seedu.duke.ui.UI;
+
+/**
+ * This class represents the "find" command by user and prints out all the tasks in the Task list containing the keyword input by user.
+ */
+public class FindCommand extends Command {
+    private String keyword;
+
+    public FindCommand(String keyword) {
+        this.keyword = keyword;
+    }
+
+    @Override
+    public void execute(TaskList tasks, UI ui, Storage storagae) {
+        TaskList matchingTasks = new TaskList();
+        String reply = "Here are the matching tasks in your list:\n\t ";
+        for (Task task: tasks) {
```

Comment:
You may want to consider abstracting out the loops so as to achieve the same level of abstraction as `ui.printReply(reply)`